examples/ui/loading_bar_widget.rs

```diff
@@ -0,0 +1,153 @@
+//! This example illustrates how to setup and use the LoadingBarWidget
+
+use bevy::{
+    prelude::*,
+    ui::widget::{LoadingBarInner, LoadingBarWidget},
+};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .insert_resource(Progress::Completed(2.0))
+        .add_startup_system(setup)
+        .add_system(update_progress_state)
+        .add_system(set_widget_progress.after(update_progress_state))
+        .add_system(update_widget_text.after(set_widget_progress))
+        .run();
+}
+
+#[derive(Resource, Clone, Copy, Debug)]
+enum Progress {
+    Loading(f32),
+    Completed(f32),
+}
+
+fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
+    let font = asset_server.load("fonts/FiraMono-Medium.ttf");
+
+    // ui camera
+    commands.spawn(Camera2dBundle::default());
+
+    // root
+    commands
+        .spawn(NodeBundle {
+            style: Style {
+                size: Size::new(Val::Percent(100.0), Val::Percent(100.0)),
+                // horizontally center children nodes
+                justify_content: JustifyContent::Center,
+                // vertically center children nodes
+                align_items: AlignItems::Center,
+                ..default()
+            },
+            background_color: Color::rgba(19.0 / 255.0, 21.0 / 255.0, 22.0 / 255.0, 1.0).into(),
```

Comment:
These colors will be much clearer as a constant.

Reply:
[`rgba_u8`](https://docs.rs/bevy/latest/bevy/prelude/enum.Color.html#method.rgba_u8) would help

Reply:
Yep, but also the named constant will help explain either their color or their purpose.